x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -80,6 +82,23 @@ export async function getFullAgentPolicy(
 
   const sourceUri = await getSourceUriForAgentPolicy(soClient, agentPolicy);
 
+  // Build up an in-memory object for looking up Package Info, so we don't have
+  // call `getPackageInfo` for every single policy, which incurs performance costs
+  const packageInfoCache = new Map<string, PackageInfo>();
```

Comment:
I am wondering if there is a chance of `packageInfo` changing in EPR, currently the cache doesn't expire, so we wouldn't return changes until kibana restart.

Reply:
Should we use an lru cache here with a limited number of entries? otherwise it could not scale at some point (lru support expiration too so it could solve what @juliaElastic mentionned)

Reply:
^ @nchaulet is correct. Every time we re-call `getFullAgentPolicy` we'll fetch the packages at that point. So, whenever we compile a policy or operate on it we'll be working with a "fresh" cache of package info for that policy.